test/e2e/framework/service_util.go

```diff
@@ -806,6 +807,108 @@ func (j *ServiceTestJig) waitForPodsReady(namespace string, pods []string) error
 	return nil
 }
 
+// CheckServiceReachability : checks when serviceType is -
+// ServiceIP: Service should be accessible from within cluster over it's -
+//  1. clusterIP and servicePort
+//  2. serviceName and servicePort
+// NodePort:
+// - Internal clients should be reachable to Service over -
+//   1. ClusterIP and servicePort
+//   2. serviceName and servicePort
+//   3. Node's internalIP and NodePort
+// - External clients should be reachable to service over -
+//   1. Node's publicIP and NodePort
+// ExternalName: FQDN of kubernetes is used as externalName(for air tight platforms). Request to externalNameService MUST resolve to IP.
+func (j *ServiceTestJig) CheckServiceReachability(f *Framework, svc *v1.Service, pod *v1.Pod) error {
+	svcEndpoints, cmds := []string{}, []string{}
+	svcType := svc.Spec.Type
+	clusterIP := svc.Spec.ClusterIP
+	ports := svc.Spec.Ports
+
+	j.SanityCheckService(svc, svcType)
+
+	switch svcType {
+	case v1.ServiceTypeClusterIP:
```

Comment:
nit: better to separate each service reachability functions like `checkClusterIPReachability` for the readability.

Reply:
I agree, I'm on it.